src/libraries/System.Net.WebSockets/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -849,6 +893,20 @@ private static int WriteHeader(MessageOpcode opcode, byte[] sendBuffer, ReadOnly
             }
         }
 
+        /// <summary>
+        /// Returns either <see cref="ValueWebSocketReceiveResult"/> or <see cref="WebSocketReceiveResult"/>.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private TResult GetReceiveResult<TResult>(int count, WebSocketMessageType messageType, bool endOfMessage)
+        {
+            if (typeof(TResult) == typeof(ValueWebSocketReceiveResult))
+            {
+                return (TResult)(object)new ValueWebSocketReceiveResult(count, messageType, endOfMessage);
```

Comment:
Could you please add a comment that this won't be boxing-unboxing because JIT is smart and will figure things out? 😊